extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/HttpConfiguration.java

```diff
@@ -49,6 +49,20 @@
     @ConfigItem(defaultValue = "8444")
     public int testSslPort;
 
+    /**
+     * If this is true then the address, scheme etc will be set from headers forwarded by the proxy server, such as
+     * {@code X-Forwarded-For}. This should only be set if you are behind a proxy that sets these headers.
+     */
+    @ConfigItem(defaultValue = "false")
+    public boolean proxyAddressForwarding;
+
+    /**
+     * If this is true and proxy address forwarding is enabled then the standard {@code Forwarded} header will be used,
+     * rather than the more common but not standard {@code X-Forwarded-For}.
+     */
+    @ConfigItem(defaultValue = "false")
+    public boolean allowForwarded;
```

Comment:
Why not use `Forwarded` if present (since it's the new preferred header) and fall-back to `X-Forwarded` otherwise? Why do we need a config?

Reply:
Because an old proxy that is using X-Forwarded likely does not understand Forwarded, and will just pass it through to the backend, allowing any remote client to spoof their address. The code I copied this from actually does this, and I think it is wrong: https://github.com/vert-x3/vertx-web/issues/1519

Reply:
Oh damn, that's subtle. OK.